src/xml_parse.rs

```diff
@@ -9,13 +9,39 @@ use xml::reader::{EventReader, XmlEvent};
 
 use super::db::{AutoType, AutoTypeAssociation, Entry, Group, Value};
 
+extern crate chrono;
+
 #[derive(Debug)]
 enum Node {
     Entry(Entry),
     Group(Group),
     KeyValue(String, Value),
     AutoType(AutoType),
     AutoTypeAssociation(AutoTypeAssociation),
+    ExpiryTime(String),
+    Expires(bool),
+}
+
+fn parse_xml_timestamp(t: &str) -> Result<chrono::NaiveDateTime> {
+    match chrono::NaiveDateTime::parse_from_str(t, "%Y-%m-%dT%H:%M:%SZ") {
+        // Prior to KDBX4 file format, timestamps were stored as ISO 8601 strings
+        Ok(ndt) => Ok(ndt),
+        // In KDBX4, timestamps are stored as seconds, Base64 encoded, since 0001-01-01 00:00:00
+        // So, if we don't have a valid ISO 8601 string, assume we have found a Base64 encoded int.
+        _ => {
+            let v = base64::decode(t).map_err(|e| Error::from(DatabaseIntegrityError::from(e)))?;
+            // Cast the Vec created by base64::decode into the array expected by i64::from_le_bytes
+            let mut a: [u8; 8] = [0, 0, 0, 0, 0, 0, 0, 0];
+            for (i, u) in v.iter().enumerate() {
+                a[i] = *u;
+            }
```

Comment:
`v` has the type `Vec<u8>`. You could use [`a.copy_from_slice(&v[0..8])`](https://stackoverflow.com/a/37669259) with a slice of `v` as an argument to avoid the for loop (not too much of an improvement, I admit)

Reply:
I will see if I can make this work and either close this or comment further if I need assistance.

I will admit to struggling a bit with converting Vec<u8> into [u8; 8]; and, I just declared success and moved on after finding something that worked.

Reply:
I've also run into the same bottleneck in the language before. The solution you have come up with is completely fine. Since you wanted some mentoring, I am being a bit nit-picky and just want to point out some opportunities for expressing things slightly more elegantly.